src/components/ThumbnailNavigation.js

```diff
@@ -78,7 +98,8 @@ export class ThumbnailNavigation extends Component {
    * Renders things
    */
   render() {
-    const { config, canvases, window } = this.props;
+    const { config, window } = this.props;
+    const { canvases } = this.state;
```

Comment:
Totally.. so previously the parent of `ThumbnailNavigation` already knew whether or not canvases were present https://github.com/ProjectMirador/mirador/commit/5eceddc8a94af54a2b93e71e3e9dd65172ddb4c7#diff-5ddc4f1cbe782222388cb5fcd2632401R92 

Now since we have moved the structure around a bit, we can't be certain that we will have canvases actually present when they are passed in. This is the case where I might want to show this element when the Window renders, but maybe the manifest hasn't finished loading.

This approach enables us to "check" that the `manifest` props coming in have access to the object to call and set canvases on. Using React's `getDerivedStateFromProps` lifecycle method we can now update our derived state from the changing `manifest` object.